sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -480,23 +498,56 @@ abstract class StringPredicate extends BinaryExpression
        false
       > SELECT _FUNC_('Spark SQL', null);
        NULL
+      > SELECT _FUNC_(encode('Spark SQL', 'utf-8'), 'Spark');
```

Comment:
let's not use this example. the input should be both string type or binary type

Reply:
BTW spark has binary literal syntax, we can use that instead of `encode`

Reply:
Done